### PR TITLE
HTTP404 -> HTTP204

### PR DESCRIPTION
**Pull request by EstherDarkish**
Uses a HTTP204 return code instead of HTTP404 when checking if a directory or element exists.

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 1]
```diff
@@ -86,10 +87,12 @@ public ResponseEntity<Flux<ElementAttributes>> getRootDirectories(@RequestHeader
     @Operation(summary = "Get if a root directory of this name exists")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "The root directory exists"),
-        @ApiResponse(responseCode = "404", description = "The root directory doesn't exist"),
+        @ApiResponse(responseCode = "204", description = "The root directory doesn't exist"),
     })
     public ResponseEntity<Mono<Void>> rootDirectoryExists(@RequestParam("directoryName") String directoryName) {
-        return ResponseEntity.ok().body(service.rootDirectoryExists(directoryName));
+        return service.rootDirectoryExists(directoryName) ?
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:93]: May be factorize out what is common ?
```suggestion
        HttpStatus status = service.rootDirectoryExists(directoryName) ? HttpStatus.OK : HttpStatus.NO_CONTENT;
        return ResponseEntity.status(status)
                    .contentType(MediaType.APPLICATION_JSON).body(Mono.empty());

```

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 2]
```diff
@@ -185,10 +188,12 @@ public ResponseEntity<Mono<Void>> notify(@PathVariable("elementUuid") UUID eleme
     @RequestMapping(method = RequestMethod.HEAD, value = "/directories/{directoryUuid}/elements/{elementName}/types/{type}")
     @Operation(summary = "Check if an element with this name and this type already exists in the given directory")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The element exists"),
-        @ApiResponse(responseCode = "404", description = "The element doesn't exist")})
+        @ApiResponse(responseCode = "204", description = "The element doesn't exist")})
     public ResponseEntity<Mono<Void>> elementExists(@PathVariable("directoryUuid") UUID directoryUuid,
                                                     @PathVariable("elementName") String elementName,
                                                     @PathVariable("type") String type) {
-        return ResponseEntity.ok().body(service.elementExists(directoryUuid, elementName, type));
+        return service.elementExists(directoryUuid, elementName, type) ?
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:195]: Same as above

**Review by laurgarn — APPROVED** [review]

**Review by bendaamerahm — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 1 above)
Comment (EstherDarkish) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java:93]: Yes, that's better.

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 2 above)
Comment (EstherDarkish) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java:195]: Yes, that's better.

**Pull request merged**